### PR TITLE
3.0 config: updated the 'Instance configuration' and 'Starting and stopping instances' topics

### LINKED ISSUE #3673: [Config] Update the 'Instance configuration' guide to using a new config
**Product:** Tarantool
**Since:** 3.0
**Root document:** 
- https://www.tarantool.io/en/doc/latest/book/admin/instance_config/
- https://www.tarantool.io/en/doc/latest/reference/tooling/tt_cli/configuration/#app-section
- https://www.tarantool.io/en/doc/latest/reference/reference_lua/other/

**SME:** @ Totktonada

See also:
- Cluster configuration: https://docs.d.tarantool.io/en/doc/3.0/concepts/configuration/#starting-tarantool-instances
- https://docs.d.tarantool.io/en/doc/3.0/concepts/configuration/configuration_code/

**Example**

Unpacked application structure (packaged using `tt pack`):

```
/home/testuser/myapp
├── bin
│   ├── tarantool
│   └── tt
├── instances.enabled
│   └── sharded_cluster -> ../sharded_cluster
├── sharded_cluster
│   ├── .rocks
│   │   └── share
│   │       └── ...
│   ├── config.yaml
│   ├── instances.yaml
│   ├── router.lua
│   ├── sharded_cluster-scm-1.rockspec
│   └── storage.lua
└── tt.yaml
```

See also: https://docs.google.com/document/d/1fOaD9_084CiQ_PrlpNrvjhrK9P3zaKgRf54fU8ZKJ4I/edit

### LINKED ISSUE #3791: [Config] Update the 'Starting and stopping instances' guide to using a new config
**Related docs. issue(s):** https://github.com/tarantool/doc/issues/3847

**Product:** Tarantool
**Since:** 3.0
**Root document:** https://www.tarantool.io/en/doc/latest/book/admin/start_stop_instance/
**SME:** @ Totktonada

**Example**

tt configuration structure:

```
└── instances.enabled
    └── app
        ├── config.yaml
        └── instances.yml
```

`config.yaml`:

```
credentials:
  users:
    replicator:
      password: 'topsecret'
      roles: [replication]
    client:
      password: 'secret'
      roles: [super]

iproto:
  advertise:
    peer: replicator@

replication:
  failover: election

groups:
  group-001:
    replicasets:
      replicaset-001:
        instances:
          instance-001:
            iproto:
              listen: 127.0.0.1:3301
          instance-002:
            iproto:
              listen: 127.0.0.1:3302
          instance-003:
            iproto:
              listen: 127.0.0.1:3303
```

`instances.yaml`:

```
app.instance-001:

app.instance-002:

app.instance-003:
```

Run app:

```
tt start app
```

### LINKED ISSUE #3847: feedback: Starting and stopping instances
**Related docs. issue(s):** https://github.com/tarantool/doc/issues/3791

**Product:** Tarantool, tt
**Since:** 3.0
**Root document (3.0 docs):** https://docs.d.tarantool.io/en/doc/3.0/concepts/configuration/#starting-tarantool-instances

**SME:** @ psergee

# Details

Depends on https://github.com/tarantool/doc/issues/3791.
Related 3.0 docs:
> <…>l – the instances configuration:
> storage1:
> storage2:
> router:
> 
> 
> 
> **|init.lua – the code of storage1 and storage2
> 
> router.init.lua – the code of router|**
> 
> 
> 
> 
> Identifying instances in code
> When the application is worki<…>

https://www.tarantool.io/en/doc/latest/book/admin/start_stop_instance/

It would be nice to see minimal examples of applications:
https://github.com/tarantool/tt/blob/master/doc/examples.md#working-with-a-set-of-instances

Without these examples the document seems incomplete.

### PR DESCRIPTION
**Pull request by andreyaksenov**
The old `Instance configuration` topic doesn't give much context on used configuration files. This PR tries to fix this issue by using a ready-to-run sample included to the documentation repository:

- https://github.com/tarantool/doc/tree/3.0/doc/code_snippets/snippets/sharding/instances.enabled/sharded_cluster

Here are the main changes:

- The `Instance configuration` topic is renamed to `Application environment` and rewritten completely. This topic now contains the info about how the application environment and layout might look in dev and prod environment. This gives some context for other topics in the `Administration` section.
   Staging: https://docs.d.tarantool.io/en/doc/3.0-instance-configuration/book/admin/instance_config/
- The `Starting and stopping instances` topic is rewritten to using the sample from the previous topic.
   Staging: https://docs.d.tarantool.io/en/doc/3.0-instance-configuration/book/admin/start_stop_instance/
   This topic also includes the content moved from other topics:
   - `Preloading Lua scripts and modules` (moved from `Instance configuration`, examples are updated to tt)
   - `Starting an instance using the tarantool command` (moved from the `Configuration topic`)
   - `tarantool command-line options` (moved from the `Configuration topic`)

Configuration:
- The `Starting Tarantool instances` section is removed from the `Configuration` topic: https://docs.d.tarantool.io/en/doc/3.0-instance-configuration/concepts/configuration/
- Updated links: https://docs.d.tarantool.io/en/doc/3.0-instance-configuration/concepts/configuration/configuration_etcd/#starting-tarantool-instances

### CODE REVIEW

File: doc/book/admin/start_stop_instance.rst [hunk 1]
```diff
@@ -3,335 +3,392 @@
...
-*   ``/var/lib/tarantool/<instance_name>/`` -- snapshots and write-ahead logs.
-*   ``/var/run/tarantool/<instance_name>.control`` -- a control socket. This is
+*   ``sharded_cluster/var/log/<instance_name>/*.log`` -- instance :ref:`logs <admin-logs>`.
+*   ``sharded_cluster/var/lib/<instance_name>/*.snap`` -- snapshots.
+*   ``sharded_cluster/var/lib/<instance_name>/*.xlog`` -- write-ahead logs.
+*   ``sharded_cluster/var/run/<instance_name>/tarantool.control`` -- a control socket. This is
     a Unix socket with the Lua console attached to it. This file is used to connect
     to the instance console.
-*   ``/var/run/tarantool/<instance_name>.pid`` -- a PID file that ``tt`` uses to
+*   ``sharded_cluster/var/run/<instance_name>/tarantool.pid`` -- a PID file that ``tt`` uses to
```
Comment (psergee) [inline doc/book/admin/start_stop_instance.rst]: Currently, `tt` does not user tarantool pid file at all. All actions performed using PID from `tt.pid` file, which contains `tt` watchdog process id. `tt` watchdog, in turn, knows PID of its child `tarantool` process.

**Review by psergee — APPROVED** [review]
Thank you for the patch! lgtm.

File: doc/book/admin/start_stop_instance.rst (hunk 1 above)
Comment (andreyaksenov) [reply doc/book/admin/start_stop_instance.rst]: thanks for the clarification, will fix

File: doc/book/admin/instance_config.rst [hunk 2]
```diff
@@ -1,177 +1,204 @@
...
 
-* [Optional] An :ref:`application file <app_server-launching_app>` with
-  instance-specific logic. Put this file into the ``/usr/share/tarantool/``
-  directory.
+The main steps of creating and preparing the application for deployment are:
 
-  For example, ``/usr/share/tarantool/my_app.lua`` (here we implement it as a
-  :ref:`Lua module <app_server-modules>` that bootstraps the database and
-  exports ``start()`` function for API calls):
+1.  :ref:`Initializing a local environment <admin-instance_config-init-environment>`.
```
Comment (p7nov) [inline doc/book/admin/instance_config.rst:14]: Not sure about "local" here. Maybe just "a tt environment"?
Local env is typical for dev time, but on real clusters the system tt mode can be used.

Comment (p7nov) [inline doc/book/admin/instance_config.rst]: Not sure about "local" here. Maybe just "a tt environment"?
Local env is typical for dev time, but on real clusters the system tt mode can be used.

File: doc/book/admin/instance_config.rst [hunk 3]
```diff
@@ -1,177 +1,204 @@
...
-     box.cfg {
-         listen = 3301;
-     }
+Initializing a local environment
+--------------------------------
 
-     -- load my_app module and call start() function
-     -- with some app options controlled by sysadmins
-     local m = require('my_app').start({...})
+Before creating an application, you need to set up a local environment for ``tt``:
```
Comment (p7nov) [inline doc/book/admin/instance_config.rst:36]: Here we can briefly say about system environments and why we choose local.

File: doc/book/admin/instance_config.rst (hunk 2 above)
Comment (p7nov) [reply doc/book/admin/instance_config.rst]: BTW, if you use the exact section name in a link, you don't need to write it:
```
:ref:`instance_config-init-environment`
```

File: doc/book/admin/instance_config.rst [hunk 4]
```diff
@@ -1,177 +1,204 @@
...
+-   ``instances.enabled``: contains a symlink to the packed ``sharded_cluster`` application.
 
-    -- preload.lua --
-    print("Preloading:")
-    print("... arg is:", ...)
-    print("Passed args:", arg[1], arg[2])
+-   ``sharded_cluster``: a packed application. In addition to files created during the application development, includes the ``.rocks`` directory containing application dependencies (for example, ``vshard``).
 
-You get the following output:
+-   ``tt.yaml``: a tt configuration file.
```
Comment (p7nov) [inline doc/book/admin/instance_config.rst]:
```suggestion
-   ``tt.yaml``: a ``tt`` configuration file.
```

File: doc/book/admin/start_stop_instance.rst [hunk 5]
```diff
@@ -3,335 +3,387 @@
 Starting and stopping instances
 ===============================
 
-To start a Tarantool instance from an :ref:`instance file <admin-instance_file>`
-using the :ref:`tt <tt-cli>` utility:
+This section describes how to manage instances in a Tarantool cluster using the :ref:`tt <tt-cli>` utility.
+A cluster can include multiple instances that run different code.
+A typical example is a cluster application that includes router and storage instances.
+For example, you can manage instances in the following ways:
```
Comment (p7nov) [inline doc/book/admin/start_stop_instance.rst]: "manage instances in the following ways" sounds a bit awkward. Maybe something more goal-oriented? Or just: "Particularly, you can perform the following actions:"

File: doc/book/admin/start_stop_instance.rst [hunk 6]
```diff
@@ -3,335 +3,387 @@
...
+*   ``/var/lib/tarantool/<instance_name>/``
+*   ``/var/run/tarantool/<instance_name>/``
+
+
+.. _configuration_run_instance_tarantool:
+
+Starting an instance using the tarantool command
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``tarantool`` command provides additional :ref:`options <configuration_command_options>` that might be helpful for development purposes.
```
Comment (p7nov) [inline doc/book/admin/start_stop_instance.rst:64]: Maybe highlight somehow that `tarantool` run _only_ specific instances?

**Review by p7nov — APPROVED** [review]
LGTM.

File: doc/book/admin/instance_config.rst (hunk 3 above)
Comment (andreyaksenov) [reply doc/book/admin/instance_config.rst:36]: According to the tt's readme, this information sounds a bit specific for the purpose of this topic. I'd better clarify this in the `Launch modes` section: https://www.tarantool.io/en/doc/latest/reference/tooling/tt_cli/configuration/#launch-modes

> When tt is installed from a repository by a package manager (apt, rpm, ...) a "system" config file (/etc/tarantool/tt.yaml) is included which forms the "system" environment - the case when tt replaces the tarantoolctl.

File: doc/book/admin/instance_config.rst (hunk 2 above)
Comment (andreyaksenov) [reply doc/book/admin/instance_config.rst]: > Not sure about "local" here. Maybe just "a tt environment"?
Local env is typical for dev time, but on real clusters the system tt mode can be used.

This section emphasise that the environment is created for development. So, I'd keep `local` or maybe changed to `development`.

File: doc/book/admin/start_stop_instance.rst (hunk 6 above)
Comment (andreyaksenov) [reply doc/book/admin/start_stop_instance.rst:64]: I think that the next sentence should give this understanding:

> Below is the syntax for starting a Tarantool instance configured in a file:

**Pull request merged**